From: Sjoerd Simons <sjoerd@beast.(none)>
Date: Tue, 15 Aug 2006 12:24:34 +0000 (+0200)
Subject: hald-runner/runner.c: Close the stderr filedescriptor after reading it to
X-Git-Tag: HAL_0_5_8
X-Git-Url: http://gitweb.freedesktop.org/?p=hal.git;a=commitdiff;h=fcb6aa0df90b7ee61a58c207896448131d3aa180

hald-runner/runner.c: Close the stderr filedescriptor after reading it to
prevent leaks. Patch by Mike Hommey
---

--- a/hald-runner/runner.c
+++ b/hald-runner/runner.c
@@ -163,6 +163,7 @@ run_exited(GPid pid, gint status, gpoint
 	if (rd->stderr_v >= 0) {
 		/* Need to read stderr */
 		error = get_string_array_from_fd(rd->stderr_v);
+		close(rd->stderr_v);
 		rd->stderr_v = -1;
 	}
 	if (rd->msg != NULL)
